Bug fix: 1. App will not crash on startup if location services are disabled on the device.

BarAgam opened the pull request:
Bug fix:
1. App will not crash on startup if location services are disabled on the device.

barmako committed:
Bug fix:
1. App will not crash on startup if location services are disabled on the device.

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/120" target="_blank"><img src="https://app.review.ninja/52010471/pull/120/badge" alt="ReviewNinja"/>

teamagam1 commented on data/src/main/java/com/teamagam/gimelgimel/data/location/LocationFetcher.java:
```diff
@@ -133,6 +137,16 @@ public void requestLocationUpdates() {
     }
 
+    @SuppressWarnings("MissingPermission")
+    private void requestLocationUpdates(String provider) {
```
this is a private method and should be after all the publics

teamagam1 reviewed (changes requested):
no one uses *isProviderExistsAndEnabled* method anymore, you can delete it.

teamagam1 commented on data/src/main/java/com/teamagam/gimelgimel/data/location/LocationFetcher.java:
```diff
@@ -133,6 +137,16 @@ public void requestLocationUpdates() {
 
+    @SuppressWarnings("MissingPermission")
+    private void requestLocationUpdates(String provider) {
+        try {
```
what happens if it cannot add any provider? does the user knows those sort of things?
what if the users enables those options after the application initiated?

barmako committed:
Bug fix:
1. App will not crash on startup if location services are disabled on the device.
